iothub/device/samples/PnpDeviceSamples/Thermostat/Program.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Client;
+using Microsoft.Azure.Devices.Shared;
+using PnpHelpers;
+
+namespace TemperatureController
+{
+    public class Program
+    {
+        private const string DeviceConnectionString = "device_connection_string_here";
+        private const string ModelId = "dtmi:com:example:TemperatureController;1";
```

Comment:
_camelCase for private properties here and everywhere?

Reply:
yes, true for private properties, but this is a const, for which we follow CamelCase

Reply:
you meant PascalCasing right? But isn't that for local variables/fields?

Reply:
yes, you are right, I meant PascalCasing. 
By convention, C# uses PascalCase for all constants; you can find some documentation here: https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions

Also, this for reference: https://stackoverflow.com/questions/242534/c-sharp-naming-convention-for-constants

Reply:
Ok. The dotnet guide you mentioned doesn't say anything about private const though.

Reply:
Also our coding [guideline ](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/doc/coding-style.md)says to follow PascalCasing for constant local variables and fields

Reply:
> Ok. The dotnet guide you mentioned doesn't say anything about private const though.

The access modifier here doesn't matter, for constant fields, we use PascalCase.
Had it not been a const, then we would have declared it as internal/ private, and used camelCase (with the appropraite prefix for static, instance var etc).
I cannot see a use-case for having a public field, that is not a const. If there is a use-case for a public field, it is recommended to expose it as a property instead (with a private backing field).

>Also our coding guideline says to follow PascalCasing for constant local variables and fields

yes, you are correct, and this is a **constant field**. This is also echoed via the [VS editorconfig](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.editorconfig#L204) that we use to enforce the C# coding style.